### ISSUE TITLE
[SCM-190] Location of cvs-settings.xml should be in user.home and not in user.dir

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#429]
**[Thibaut CAZENAVE](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=thibautcazenave)** opened **[SCM-190](https://issues.apache.org/jira/browse/SCM-190?redirect=false)** and commented

In static method getSettings() of CvsUtil class (org.apache.maven.scm.provider.cvslib.util), we try to get the cvs-settings.xml property file from ${user.dir}/.scm/

---

File scmUserDir = new File( System.getProperty( "user.dir" ), ".scm" );
File settingsFile = new File( scmUserDir, "cvs-settings.xml" );

---

The SCM documentation (http://maven.apache.org/scm/cvs.html) say this file should be in ${user.home}/.scm/ and not {user.dir}/.scm/.

This should be easy to correct.

Thibaut

---

**Affects:** 1.0-beta-3

**Comment by jira-importer** [comment]
**[Emmanuel Venisse](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=evenisse)** commented

Fixed

**Comment by jira-importer** [comment]
**[Eric Bernstein](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ebernstein)** commented

Any chance of 1.0 being released soon?  I'd love to see this fix out in a release, and there hasn't been an scm release in a very long time.

Just a thought though - would it not be better if cvs-settings.xml was in ${user.home}/.m2/scm/ so all the maven config files were grouped together?

**Comment by jira-importer** [comment]
**[Emmanuel Venisse](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=evenisse)** commented

I hope to release it for the end of the year.
I don't want to store it in .m2/scm because maven-scm is independant of maven

**Comment by jira-importer** [comment]
**[Joerg Schaible](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=joehni)** commented

You realize, that this sounds slightly strange for a component using a "maven-" prefix in its name? ;-)
Personally I'd rather decrease the amount of directories and files cluttering my home directory. Even if maven-scm is independent of Maven 95% of the users are not aware of this and will search for those settings in .m2 ...

**Comment by jira-importer** [comment]
**[Emmanuel Venisse](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=evenisse)** commented

it is used by maven2, maven1 continuum, so we can't use .m2 directory

**Comment by jira-importer** [comment]
**[Eric Bernstein](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ebernstein)** commented

The .scm directory doesn't seem all that bad - esp given that it's shared across projects.

Is there anything that needs to be done to expedite the scm release?  My company's cvs will actually hang when using compression - which is another issue entirely, but having a good way to configure that would be incredibly useful.  If there are tasks that need to be done before scm can be released I'm sure I can find some cycles to help out.  Let me know if there's anything I can do to get this out sooner.  Thanks.

**Comment by jira-importer** [comment]
**[Emmanuel Venisse](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=evenisse)** commented

It's my latest message there because it's better to discuss on mailing list ;) but the roadmap is there : http://jira.codehaus.org/browse/SCM?report=com.atlassian.jira.plugin.system.project:roadmap-panel
